### DISCUSSION TITLE
关于Local图源的刮削器/Scrapers for local picture sources

### DISCUSSION THREAD

**Discussion by tiehu** [koishijs/koishi-plugin-booru#101] [Q&A] [answered]
koishi-plugin-booru-local version: 1.0.0 (get from the koishi market)

我不太确定这个功能是否已经开发完成。经过反复测试，我注意到这个功能似乎与文档中描述的有很大的不同。
文档中给出的示例配置为：#name#{filename}-{tag}
我原本以为按照这个配置，图片文件名应该为：filename-tag.png
我甚至还尝试了：{filename}-{tag}.png
但最后，通过查阅源代码和反复测试，我发现它期望的格式似乎是：filename-[tag].png

在将文件名格式设置为“filename-[tag].png”后，我检查了data/booru-local/booru-maps.json文件，确认其中的“tag”字段已经配置为预期中的字符串。
然而，当我实际使用时，发现bot并不会针对标签进行选择，而是直接从全部本地图片中随机抽取一张发送。

我希望知道如何解决这个问题，万分感谢。

The following is from machine translation:

I'm not sure if this feature has been developed yet. After repeated testing, I noticed that this feature seemed to be quite different from what was described in the documentation.

The example configuration given in the documentation is: #name#{filename}-{tag}

I originally thought that with this configuration, the image filename should be: filename-tag.png

I even tried: {filename}-{tag}.png

But in the end, through looking at the source code and repeated testing, I found that the desired format seems to be: filename-[tag].png

After setting the filename format to "filename-[tag].png", I checked the data/booru-local/booru-maps.json file to confirm that the "tag" field in it was configured as the expected string.

However, when I actually used it, I found that the bot did not select labels, but just randomly selected one from all the local images and sent it.

I'd like to know how to solve this problem. Thank you very much.

**Comment by Lipraty** [comment]
Indeed, this is a feature that has been overlooked. However, I will try to improve it when I have some spare time. 
This should be an FR issue, and I will move it to the appropriate place.

是的，这是一个被忽视的功能。不过，当我有时间时，我会尝试改进。
这应该是一个 `issue`，我会把它移到合适的地方。

**Comment by Lipraty** [comment, marked as answer]
The bug in the local image source related to tags has been fixed, and the documentation has been updated to more accurately describe how to use the scraper. (Waiting for merge) 😄 

本地图像源中与标签相关的错误已得到修复，并且文档已更新以更准确地描述如何使用抓取工具。
